libbpf-rs/src/map.rs

```diff
@@ -295,6 +364,72 @@ impl Map {
             )));
         };
 
+        self.update_raw(key, value, flags)
+    }
+
+    /// Update an element in an per-cpu map with one value per cpu.
+    ///
+    /// `key` must have exactly [`Map::key_size()`] elements. `value` must have one
+    /// element per cpu (see [`num_possible_cpus()`]) with exactly [`Map::value_size()`]
+    /// elements each.
+    ///
+    /// For per-cpu maps, [`Map::update_percpu()`] must be used.
+    pub fn update_percpu(
+        &mut self,
+        key: &[u8],
+        values: &Vec<Vec<u8>>,
+        flags: MapFlags,
+    ) -> Result<()> {
+        if !self.is_percpu() && self.map_type() != MapType::Unknown {
+            return Err(Error::InvalidInput(format!(
+                "update() must be used for maps that are not per-cpu (type of the map is {})",
+                self.map_type(),
+            )));
+        }
+
+        if values.len() != num_possible_cpus()? {
+            return Err(Error::InvalidInput(format!(
+                "number of values {} != number of cpus {}",
+                values.len(),
+                num_possible_cpus()?
```

Comment:
Cache value of `num_possible_cpus()`?

Reply:
I agree that it looks a bit expensive to call `num_possible_cpus()` everytime. But I just took a look at the implementation of `libbpf_num_possible_cpus()` in libbpf itself and it already caches the result (https://github.com/libbpf/libbpf/blob/e671a47bc279c6a73d1fee5aa522b0979fd1a33a/src/libbpf.c#L10905). So I think another layer of caching is not necessary.